### ISSUE TITLE
Should not let users export PYTHONHOME for PyTorch example local mode

### ISSUE THREAD

**Issue by hkvision** [intel-analytics/analytics-zoo#2690]
I run pytorch/train/mnist/main.py on Spark local and get the following error after the training:
```
java.lang.IllegalArgumentException: requirement failed
	at scala.Predef$.require(Predef.scala:212)
	at com.intel.analytics.zoo.feature.PythonFeatureSet$.toArrayTensor(FeatureSet.scala:390)
	at com.intel.analytics.zoo.pipeline.api.net.TorchModel.getExtraParameter(TorchModel.scala:188)
	at com.intel.analytics.zoo.pipeline.api.keras.models.InternalOptimizerUtil$$anonfun$8.apply(Topology.scala:1041)
	at com.intel.analytics.zoo.pipeline.api.keras.models.InternalOptimizerUtil$$anonfun$8.apply(Topology.scala:1041)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:394)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.generic.Growable$class.$plus$plus$eq(Growable.scala:59)
	at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:104)
	at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:48)
	at scala.collection.TraversableOnce$class.to(TraversableOnce.scala:310)
	at scala.collection.AbstractIterator.to(Iterator.scala:1334)
	at scala.collection.TraversableOnce$class.toBuffer(TraversableOnce.scala:302)
	at scala.collection.AbstractIterator.toBuffer(Iterator.scala:1334)
	at scala.collection.TraversableOnce$class.toArray(TraversableOnce.scala:289)
	at scala.collection.AbstractIterator.toArray(Iterator.scala:1334)
	at org.apache.spark.rdd.RDD$$anonfun$take$1$$anonfun$29.apply(RDD.scala:1364)
	at org.apache.spark.rdd.RDD$$anonfun$take$1$$anonfun$29.apply(RDD.scala:1364)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Also:
1. Is the following warning expected? Look quite like an error stack actually:
```
Warn: jep.JepException: <class 'NameError'>: name 'loader3b553ae7_0_iter_true' is not defined
	at <string>.<module>(<string>:2)
	at jep.Jep.exec(Native Method)
	at jep.Jep.exec(Jep.java:478)
	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply$mcV$sp(PythonInterpreter.scala:94)
	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:93)
	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:93)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Warn: jep.JepException: <class 'StopIteration'>
	at jep.Jep.exec(Native Method)
	at jep.Jep.exec(Jep.java:478)
	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply$mcV$sp(PythonInterpreter.scala:94)
	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:93)
	at com.intel.analytics.zoo.common.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:93)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

2. Seems if JEP is not installed, the program would crash? Can we instead raise some error on Python side directly?

**Comment by jason-dai** [comment]
Yes, we need to improve the error checking and error message

**Comment by qiuxin2012** [comment]
The error is thrown after training&validation finished, when optimizer try to get extraparameters.
This bug is fixing in https://github.com/intel-analytics/analytics-zoo/pull/2630.

**Comment by hkvision** [comment]
Four issues in total:
- `java.lang.IllegalArgumentException: requirement failed` showed above.
- In the README: https://github.com/intel-analytics/analytics-zoo/tree/master/pyzoo/zoo/examples/pytorch/train/mnist
```
conda install pytorch-cpu torchvision-cpu -c pytorch #command for linux
```
This will install pytorch 1.1.0 but not 1.5.0 as guided in the readme.
- Not installing JEP would cause the program to crash. Better detect first on Python side.
- [Optional] The warning stack of JEP may not be friendly to users, seems like to be an error stack.

**Comment by hkvision** [comment]
Better fix the issue of letting users export PYTHONHOME for running PyTorch example in local mode. This environment variable would affect other programs.
If I first run on local mode and export PYTHONHOME, then I want to run for yarn-client mode without starting a new session or cleaning the environment variables, it would throw the following error:
```
Start to pack current python env
Collecting packages...
CondaPackError: Failed to determine path to environment: 'orca-kai'. This may be due to conda not being on your PATH. The full error is below:

b''
```

**Comment by qiuxin2012** [comment]
Fixed in #2869
